libstuff/SSSLState.cpp

```diff
@@ -22,26 +22,45 @@ SSSLState::~SSSLState() {
 }
 
 // --------------------------------------------------------------------------
-SSSLState* SSSLOpen(int s, SX509* x509) {
+SSSLState* SSSLOpen(int s) {
     // Initialize the SSL state
     SASSERT(s >= 0);
     SSSLState* state = new SSSLState;
     state->s = s;
 
-    mbedtls_ctr_drbg_seed(&state->ctr_drbg, mbedtls_entropy_func, &state->ec, 0, 0);
-    mbedtls_ssl_config_defaults(&state->conf, MBEDTLS_SSL_IS_CLIENT, MBEDTLS_SSL_TRANSPORT_STREAM, 0);
+    // All new code here.
+    mbedtls_ssl_init(&state->ssl);
+    mbedtls_ssl_config_init(&state->conf);
+    mbedtls_net_init(&state->net_ctx);
+    state->net_ctx.fd = s;
 
-    mbedtls_ssl_setup(&state->ssl, &state->conf);
+    mbedtls_entropy_init(&state->ec);
+    mbedtls_ctr_drbg_init(&state->ctr_drbg);
+    mbedtls_ctr_drbg_seed(&state->ctr_drbg, mbedtls_entropy_func, &state->ec, nullptr, 0);
+
+    if (mbedtls_ssl_config_defaults(&state->conf, MBEDTLS_SSL_IS_CLIENT, MBEDTLS_SSL_TRANSPORT_STREAM, MBEDTLS_SSL_PRESET_DEFAULT)) {
+        STHROW("ssl config defaults failed");
+    }
 
     mbedtls_ssl_conf_authmode(&state->conf, MBEDTLS_SSL_VERIFY_OPTIONAL);
```

Comment:
I think we should set this to verify, why would we not want to verify the SSL cert is correct? That kinda defeats the entire purpose.

Reply:
See comment below.